src/components/Nav/Nav.js

```diff
@@ -8,34 +8,79 @@ import './Nav.scss';
 const Nav = () => {
   const [navbarOpen, setNavbarOpen] = useState(false);
   const [drawerOpen, setDrawerOpen] = useState(false);
+  const [loggedIn, setLoggedIn] = useState(false);
+
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+    token ? setLoggedIn(true) : setLoggedIn(false);
+  }, [loggedIn]);
```

Comment:
```suggestion
  const token = localStorage.getItem('token');
```

이 과정은 token 값 하나로 관리가 가능합니다.
token 변수는 렌더링 될 때마다 다시 계산을 해서 null 혹은 token에 담겨있는 값이 들어가게 됩니다.
이 값은 완벽하게 true 나 false의 값은 아니지만 truthy 혹은 falsy 값이라 해서 마치
true나 false 처럼 사용할 수 있는 값입니다.

그래서 저 한줄의 코드로 동일한 동작을 구현할 수 있겠습니다.
truthy와 falsy는 아래 블로그 링크를 참고해보세요
https://learnjs.vlpt.us/useful/02-truthy-and-falsy.html